tensorflow/compiler/xla/service/gpu/cudnn_batchnorm_thunk.cc

```diff
@@ -30,50 +31,45 @@ namespace gpu {
 
 namespace dnn = se::dnn;
 
-static std::pair<dnn::BatchDescriptor /*input_desc*/,
-                 dnn::BatchDescriptor /*scale_offset_desc*/>
-MakeDescriptors(const Shape& shape, int64 feature_index) {
-  std::vector<int64> logical_to_physical =
-      LayoutUtil::MakeLogicalToPhysical(shape.layout());
-
-  auto physical_dim_size = [&](int64 physical_dim) {
-    return shape.dimensions(LayoutUtil::Major(shape.layout(), physical_dim));
-  };
-
-  // Batchnorm only cares about the location of the depth (aka "feature") dim.
-  // The other dims are all treated the same.  Thus we can use the kBatchDepthYX
-  // cudnn layout for any XLA shape+layout, even XLA shapes that don't have
-  // exactly 4 dimensions: We put everything that comes before the feature dim
-  // into "batch", and everything that comes after the feature dim into "Y".
-  int64 batch_size = 1;
-  int64 y_size = 1;
-  int64 physical_dim;
-  for (physical_dim = 0; physical_dim != logical_to_physical[feature_index];
-       ++physical_dim) {
-    CHECK_LT(physical_dim, shape.dimensions_size());
-    batch_size *= physical_dim_size(physical_dim);
+namespace {
+void CheckInputOutputPrimitivetypeValid(const HloInstruction* hlo) {
+  // Check Inputs.
+  int64 num_operands = hlo->operand_count();
+  PrimitiveType operand_dtype = hlo->operand(0)->shape().element_type();
```

Comment:
In XLA we just say type or "primitive type" instead of dtype.

Reply:
Sorry for the oversight. I missed that one.